grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -283,6 +287,95 @@ class StockMovementController {
         render(template: "documents", model: [stockMovement: stockMovement])
     }
 
+    def comments = {
+        def stockMovement = getStockMovement(params.id)
+        render(template: "comments", model: [stockMovement: stockMovement])
+    }
+
+    def addComment = {
+        def stockMovement = getStockMovement(params.id)
+        List<User> recipients = User.list()
+        return [stockMovement: stockMovement, comment: new Comment(), recipients: recipients]
+    }
+
+    def editComment = {
+        def stockMovement = getStockMovement(params['stockMovement.id'])
+        if (!stockMovement) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'stockMovement.label', default: 'Stock Movement'), params['stockMovement.id']])}"
+            redirect(action: "list")
+            return
+        }
+        Comment comment = Comment.get(params?.id)
+        if (!comment) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'comment.label', default: 'Comment'), comment.id])}"
+            redirect(action: "show", id: stockMovement?.id)
+            return
+        }
+        if (comment.sender.id != session.user.id) {
+            flash.message = "${g.message(code: 'auth.notAuthorized.message')}"
+            redirect(action: "show", id: stockMovement?.id)
+            return
+        }
+        render(view: "addComment", model: [stockMovement: stockMovement, comment: comment])
+    }
+
+    def deleteComment = {
+        def stockMovement = getStockMovement(params['stockMovement.id'])
+        if (!stockMovement) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'stockMovement.label', default: 'Stock Movement'), params['stockMovement.id']])}"
+            redirect(action: "list")
+            return
+        }
+        Comment comment = Comment.get(params?.id)
+        if (!comment) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'comment.label', default: 'Comment'), params.id])}"
+            redirect(action: "show", id: stockMovement?.id)
+            return
+        }
+
+        if (comment.sender.id != session.user.id) {
+            throw new AccessDeniedException("You are not authorized to perform this action")
+        }
+
+        stockMovement.requisition?.removeFromComments(comment)
+        if (!stockMovement.requisition.hasErrors() && stockMovement.requisition?.save()) {
+            flash.message = "${g.message(code: 'default.comments.deleted.label')}"
```

Comment:
Well I think you technically not deleted the comment yet (you only removed it from requisition so far)

Reply:
I agree, and I think this should be handled by the `all-delete-orphan` in the Requisition domain.

Reply:
@kchelstowski To clarify, you mean we should only use `comment.delete()` and let the all-delete-orphan handle removing the comments from the association, correct?

Reply:
@jmiranda I meant that doing `stockMovement.requisition.removeFromComments(comment)` should be enough if we use the orphan removal.
Setting orphan on the comment would mean, that a comment's existence "does not have sense" if it doesn't have requisition associated, so it's gonna be deleted whenever we unassign a comment from a requisition.

tl;dr - calling `comment.delete()` is not neccesary if we use the `all-delete-orphan`. `requisition.removeFromComments` should do the job.

Reply:
you are right, `Requisiton` is missing `cascade: "all-delete-orphan"`